pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/CreateSubscriptionTest.java

```diff
@@ -127,4 +139,86 @@ public void createSubscriptionOnPartitionedTopicWithPartialFailure() throws Exce
+        Awaitility.await().untilAsserted(() -> {
+            assertEquals(subscription.getCursor().getMarkDeletedPosition().getLedgerId(), messageId.getLedgerId());
+            assertEquals(subscription.getCursor().getMarkDeletedPosition().getEntryId(), messageId.getEntryId());
+        });
+        properties = subscription.getSubscriptionProperties();
+        assertTrue(properties.containsKey("1"));
+        assertTrue(properties.containsKey("2"));
+        assertEquals(properties.get("1"), "1");
+        assertEquals(properties.get("2"), "2");
+
+        consumer.close();
+        producer.close();
+        // restart broker, consumer use old properties
+        restartBroker();
+
+        Consumer<byte[]> consumer2 = pulsarClient.newConsumer().topic(topic).receiverQueueSize(1)
+                .subscriptionProperties(map).subscriptionName(subName).subscribe();
+        PersistentSubscription subscription2 = (PersistentSubscription) pulsar.getBrokerService()
+                .getTopicReference(topic).get().getSubscription(subName);
+        Awaitility.await().untilAsserted(() -> {
+            Map<String, String> properties2 = subscription2.getSubscriptionProperties();
+            assertTrue(properties2.containsKey("1"));
+            assertTrue(properties2.containsKey("2"));
+            assertEquals(properties2.get("1"), "1");
+            assertEquals(properties2.get("2"), "2");
+        });
+        consumer2.close();
+
+        // create a new consumer with new properties, the new properties should be ignored
+        Map<String, String> map3 = new HashMap<>();
+        map3.put("3", "3");
+        map3.put("4", "4");
+        Consumer<byte[]> consumer3 = pulsarClient.newConsumer().topic(topic).receiverQueueSize(1)
+                .subscriptionProperties(map3).subscriptionName(subName).subscribe();
+        Map<String, String> properties3 = subscription.getSubscriptionProperties();
+        assertTrue(properties3.containsKey("1"));
+        assertTrue(properties3.containsKey("2"));
+        assertEquals(properties3.get("1"), "1");
+        assertEquals(properties3.get("2"), "2");
+        consumer3.close();
+
+        //restart and create a new consumer with new properties, the new properties should be updated
+        restartBroker();
+        Consumer<byte[]> consumer4 = pulsarClient.newConsumer().topic(topic).receiverQueueSize(1)
+                .subscriptionProperties(map3).subscriptionName(subName).subscribe();
+        PersistentSubscription subscription4 = (PersistentSubscription) pulsar.getBrokerService()
+                .getTopicReference(topic).get().getSubscription(subName);
+        Map<String, String> properties4 = subscription4.getSubscriptionProperties();
+        assertTrue(properties4.containsKey("3"));
+        assertTrue(properties4.containsKey("4"));
+        assertEquals(properties4.get("3"), "3");
+        assertEquals(properties4.get("4"), "4");
+        consumer4.close();
+    }
```

Comment:
1. add the case: consumer subscribe without subscriptionProperties set, it will get the old one. And then restart broker, it won't get any properties?
2.  restart  broker and create a new consumer with new properties, the properties will be updated. Do not restart broker and create a new consumer with new properties, the properties will be ignored. Whether the broker restart or not will not be known for consumer, does this action will confuse the consumer?

Reply:
Unit test has been added

The configuration of the Consumer is usually the same. Even if the Broker restarts and the Consumer reconnects with the original configuration, the Properties will be restored.